### DISCUSSION TITLE
Better development experience for node SSR deployment adapter

### DISCUSSION THREAD

**Discussion by smnbbrv** [withastro/roadmap#344] [Proposal]
It would be great to give more support on development experience with "deployment adapters", e.g. with custom express server. Running full build each time when the .astro content is changed to make it work on a custom server is quite slow. 

First, one needs to rebuild the whole website.

Second, one needs to restart the node / express server.

E.g. next.js somehow manages this without needing to rebuild the custom server at all, while watching the changes on its own.

The currently available options I could find are

- making [two watches](https://github.com/withastro/rfcs/discussions/273#discussioncomment-3891278)
- reimport the module every time the middleware is requested, see below

```ts
import express from 'express';

const app = express();

app.use(express.static('dist/client/'));

app.use((...args) => {
    // clear import cache every time for reloading in dev mode
    import('../dist/server/entry.mjs' + '?_=' + Date.now()).then(({ handler }) => {
        handler(...args);
    });
});

app.listen(8080);
```

Both aren't very pleasant. First one because of nodemon-chaining (when astro content is changed, it gets rebuilt and then it triggers the rebuild of the express server). Second one has no chaining required (however still has two nodemons), but introduces dev-only logic into the application, that can potentially affect it in production, and most likely can introduce memory leaks.

**Comment by florian-lefebvre** [comment]
Closing because this is outdated

**Reply by smnbbrv** [reply]
Why is this outdated? Is it possible in newer Astro versions?

**Reply by florian-lefebvre** [reply]
I was doing a purge of the oldest proposals but I'm happy to reopen! + I have an idea for this

**Reply by maxaon** [reply]
I'm also trying to integrate Astro to app, but need to build is killing dev experience.
